bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/transactions/TransactionResourceManager.java

```diff
@@ -186,6 +193,83 @@ private String getTransactionLogDirectory() {
         }
     }
 
+    /**
+     * This method gets the user specified config for the transaction auto commit timeout. Default is 120.
+     *
+     * @return int transaction auto commit timeout value
+     */
+    public int getTransactionAutoCommitTimeout() {
+        VariableKey transactionAutoCommitTimeoutKey = new VariableKey(TRANSACTION_PACKAGE_ID,
+                TRANSACTION_AUTO_COMMIT_TIMEOUT_KEY, PredefinedTypes.TYPE_INT, false);
+        Object value = ConfigMap.get(transactionAutoCommitTimeoutKey);
+        if (value == null) {
+            return DEFAULT_TRX_AUTO_COMMIT_TIMEOUT;
+        } else {
+            int transactionAutoCommitTimeout;
+            if (value instanceof Long) {
+                transactionAutoCommitTimeout = ((Long) value).intValue();
```

Comment:
Instead of casting, we can use the new feature in java 17. 
```
if (value instanceof Long value) {
    transactionAutoCommitTimeout = value.intValue();
}
```